src/Data/ArrayBuffer/ArrayBuffer.js

```diff
@@ -6,31 +6,41 @@ exports.create = function(s) {
   return function () {
     return new ArrayBuffer(s);
   };
-}
+};
 
 exports.byteLength = function(a) {
   return a.byteLength;
-}
+};
 
 exports.sliceImpl = function(s, e, a) {
   return function () {
     return a.slice(s, e);
   };
-}
+};
 
 exports.fromArray = function(s) {
   return (new Uint8Array(s)).buffer;
-}
+};
 
 exports.fromIntArray = function(s) {
   return (new Uint8Array(s)).buffer;
-}
+};
 
 exports.fromString = function(s) {
-  var l = s.length;
-  var ab = new ArrayBuffer(l * 2);
-  var a = new Uint16Array(ab);
-  for (var i = 0; i < l; i++)
-    a[i] = s.charCodeAt(i);
-  return a.buffer;
-}
+  var buf = new ArrayBuffer(s.length*2);
+  var bufView = new Uint16Array(buf);
+  for (var i=0, strLen=s.length; i<strLen; i++) {
+    bufView[i] = s.charCodeAt(i);
+  }
+ return buf;
+};
+
+exports.decodeToString = function(buffer) {
```

Comment:
Should this return a `Maybe`? What if you pass a 1 byte array?

Reply:
Good catch, yes. it absolutely should. An earlier revision did infact have a maybe, and I somehow forgot about that case as I went about simplifying.